pkg/testframework/console.go

```diff
@@ -147,3 +147,65 @@ func MustRemoveConsole(t *testing.T, client *Clientset) error {
 	}
 	return nil
 }
+
+func MustNormalLogLevel(t *testing.T, client *Clientset) error {
+	cr, err := client.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("checking if console operator LogLevel is set to 'Normal'...")
+	if cr.Spec.LogLevel == operatorsv1.Normal {
+		return nil
+	}
+	err = SetLogLevel(t, client, operatorsv1.Normal)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return nil
+}
+
+func SetLogLevel(t *testing.T, client *Clientset, logLevel operatorsv1.LogLevel) error {
+	cr, err := client.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	deployment, err := GetConsoleDeployment(client)
+	if err != nil {
+		return err
+	}
+	currentDeploymentGeneration := deployment.ObjectMeta.Generation
+	currentConfigGeneration := cr.ObjectMeta.Generation
+
+	t.Logf("setting console operator to '%s' LogLevel ...", logLevel)
+	_, err = client.Consoles().Patch(consoleapi.ConfigResourceName, types.MergePatchType, []byte(fmt.Sprintf(`{"spec": {"logLevel": "%s"}}`, logLevel)))
+	if err != nil {
+		return err
+	}
+
+	err = wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		newCr, err := client.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+		deployment, err := GetConsoleDeployment(client)
```

Comment:
This makes me wonder if it would be handy to just have a separate `GenerationChanged(resourceGeneration)` or `CompareGenerations(oldGen, newGen)` function.  I imagine we could use this a good bit in tests, though I don't think we can make it generic to all resources (strongly typed).